File: app/src/main/java/com/example/newsfeed/NewsApp/util/Resource.java
package com.example.newsfeed.NewsApp.util;

public abstract class Resource<T> {
    private final T data;
    private final String message;

    public Resource(T data, String message) {
        this.data = data;
        this.message = message;
    }

    public static final class Success<T> extends Resource<T>{
private T data;

        public Success( String message, T data) {
              super(data, message);
            this.data = data;
        }
    }
    public static final class Error<T> extends Resource<T>{
        private String message;
        private T data;
        public Error( String message, T data) {
            super(data, message);
            this.message = message;
            this.data=data;
        }
    }
    public static final class Loading<T> extends Resource<T>
    {
        public Loading(T data, String message) {
            super(data, message);
        }
    }




}

File: app/src/main/java/com/example/newsfeed/NewsApp/util/Resource.kt
package com.example.newsfeed.NewsApp.util

abstract class Resource<T>(private val data: T, private val message: String?) {
    class Success<T>(message: String?, private val data: T) : Resource<T>(data, message)
    class Error<T>(private val message: String?, private val data: T) : Resource<T>(data, message)
    class Loading<T>(data: T, message: String?) : Resource<T>(data, message)
}